apm-sniffer/apm-sdk-plugin/jedis-plugins/jedis-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jedis/v4/ConnectionProviderConstructorInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *   Licensed to the Apache Software Foundation (ASF) under one or more
+ *   contributor license agreements.  See the NOTICE file distributed with
+ *   this work for additional information regarding copyright ownership.
+ *   The ASF licenses this file to You under the Apache License, Version 2.0
+ *   (the "License"); you may not use this file except in compliance with
+ *   the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.skywalking.apm.plugin.jedis.v4;
+
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import redis.clients.jedis.HostAndPort;
+
+public class ConnectionProviderConstructorInterceptor implements InstanceConstructorInterceptor {
+    @Override
+    public void onConstruct(final EnhancedInstance objInst, final Object[] allArguments) throws Throwable {
+        if (objInst.getSkyWalkingDynamicField() != null) {
+            return;
+        }
+        Object arg = allArguments[0];
+        if (arg instanceof Iterable<?>) {
+            Iterable<?> iterable = (Iterable<?>) arg;
+            StringBuilder sb = new StringBuilder();
+            for (Object o : iterable) {
+                sb.append(o.toString()).append(",");
```

Comment:
I think you should use `StringUtil#join`.

Reply:
`StringUtil#join` doesn't support `Iterable` as parameter.

Reply:
I think it can support that easily.

Reply:
I have used `Collection<?>` instead of `Iterable<?> `, and then converted the collection to String array, that can be directly applied to `StringUtil#join(final char delimiter, final String... strings)`. Anyway, the type converting is required